rfcs/react-components/components/Spinner.md

```diff
@@ -0,0 +1,147 @@
+| Lightning Design System | `Spinner`          | [Spinner](https://react.lightningdesignsystem.com/components/spinners/)                   | A more robust indeterminate Spinner with props that allow the Spinner to be rendered in a container, add styling to said container, inline the Spinner, and change the size                                                                                                                       |
+
+### Comparison of v8 and v0
+
+The existing components are:
+
+- v8
+  - `Spinner`
+- v0
+  - `Loader`
+- Both `Spinner` and `Loader` have very similar props
+- `Loader` has `delay`, while `Spinner` has aria props exposed for accessibility.
+
+## Sample Code
+
+Basic examples:
+
+```jsx=
+import { Spinner } from "@fluentui/react-spinner";
+
+function App() {
+    return (
+        <Spinner size="medium" label="Loading" />
+    )
+}
+
+```
+
+## Variants
+
+- Determinate Spinner
+  - Add a `value` prop that prefills the `Spinner` to a specific percentage. Can be implemented using a value from 0 to 1.
+- Indeterminate Spinner
+  - The default Spinner that spins around indefinitely
+
+### Shape
+
+The Spinner is represented as a circle with an arc of a darker shade rotating the circumfrence
+
+## API
+
+From [Spinner.types.tsx]()
+
+### Slots
+
+- `root` - The root element of the Spinner. The html element is a `span`
+- `loader` - The svg element that gets animated into a Spinner. The html element is `svg`
+- `label` - The text shown with the Spinner. This uses the `Text` control
+
+### Props
+
+```jsx
+export type SpinnerProps = ComponentProps & React.HTMLAttributes<HTMLElement> & {
+  /* The appearance of the Spinner*/
+  appearance?: "primary" | "inverted";
+  /* ariaLabel prop for accessibility */
+  ariaLabel?: string;
+  /* The label prop allows user to add text. Defaults to "Loading..." */
+  label?: <Text />;
+  /* The labelPosition prop allows user to set the location of the label*/
+  labelPosition?: "above" | "below" | "before" | "after";
+  // new prop for vertical? : boolean. Will allow for the labelPosition prop to only need to be "before" and "after"
+  vertical?: boolean;
+  /* The size prop sets the size of the Spinner
+   * @defaultValue "medium"*/
+  size?: "tiny" | "x-small" | "small" | "medium" | "large" | "x-large" | "huge";
+  // inactive ? : boolean
+  status?: "active" | "success" | "error" | "inactive";
+  /** The indeterminate prop sets whether the Spinner is a determinate Spinner or indetermindate Spinner*/
+  indeterminate?: boolean;
+};
+```
+
+## Structure
+
+```html
+<span class="root">
```

Comment:
for a determinate spinner we should probably have `role="progressbar"` the aria values:

* aria-valuenow=
* aria-valuemin
* aria-valuemax